cmake/modules/AddSwift.cmake

```diff
@@ -198,38 +201,69 @@ function(_add_variant_c_compile_flags)
     RESULT_VAR_NAME result)
 
   is_build_type_optimized("${CFLAGS_BUILD_TYPE}" optimized)
-  if(optimized)
-    list(APPEND result "-O2")
 
-    # Add -momit-leaf-frame-pointer on x86.
-    if("${CFLAGS_ARCH}" STREQUAL "i386" OR "${CFLAGS_ARCH}" STREQUAL "x86_64")
-      list(APPEND result "-momit-leaf-frame-pointer")
+  # MSVC has different options to most compilers!
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    list(APPEND result "-DLLVM_ON_WIN32")
+    list(APPEND result "-D_CRT_SECURE_NO_WARNINGS")
+    # TODO(compnerd) handle /MT
+    list(APPEND result "-D_DLL")
```

Comment:
The issue here was that I get many warnings when linking using MSVC. Something like
- `no such option -DLLVM_ON_WIN32`
- `no such option -D_CRT_SECURE_NO_WARNINGS`
- `no such option -D_DLL`

However, if we add these to the compiler flags instead of both the compiler and linker flags, we remove these warnings.

> This should be based on the target
I don't fully understand this, sorrry